rosie/src/main/java/com/karumi/rosie/view/PresenterLifeCycleLinker.java

```diff
@@ -78,13 +87,63 @@ public void pausePresenters() {
     }
   }
 
+  /**
+   * Destroys all the already registered presenters lifecycle.
+   */
   public void destroyPresenters() {
     for (RosiePresenter presenter : presenters) {
       presenter.destroy();
     }
   }
 
+  /**
+   * Registers a presenter instance.
+   *
+   * @param presenter to be registered
+   */
   public void registerPresenter(RosiePresenter presenter) {
+    if (presenter == null) {
+      throw new IllegalArgumentException("The presenter instance to be registered can't be null");
+    }
     presenters.add(presenter);
   }
+
+  /**
+   * Updates the view instance associated to all the already registered presenters.
+   *
+   * @param view to be assigned to the presenters.
+   */
+  public void setView(RosiePresenter.View view) {
```

Comment:
Yup, I understand, I just couldn't think of any case where I'd want to change the view mid-lifecycle but I'm ok if someone uses it